apm-agent-plugins/apm-apache-httpclient/apm-apache-httpclient5-plugin/src/main/java/co/elastic/apm/agent/httpclient/v5/helper/RequestChannelWrapper.java

```diff
@@ -99,7 +101,13 @@ public void sendRequest(HttpRequest httpRequest, EntityDetails entityDetails, Ht
                     if (host != null) {
                         span.appendToName(host);
                     }
-                    span.getContext().getHttp().withMethod(method).withUrl(httpRequest.getRequestUri());
+                    String requestUri = httpRequest.getRequestUri();
+                    // starting with httpclient 5.1 "getRequestUri" can return relative path, not always the absolute one
+                    // thus we have to fallback on using the URI instance
+                    if(requestUri != null && requestUri.startsWith("/") && httpRequestURI != null) {
+                        requestUri = httpRequestURI.toString();
+                    }
+                    span.getContext().getHttp().withMethod(method).withUrl(requestUri);
```

Comment:
[for reviewer] this is another API change that is only problematic when trying to 5.4.4 version and later.